.github/workflows/terraform_test.yaml

```diff
@@ -0,0 +1,118 @@
+name: Terraform Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      module_to_test:
+        description: |
+          The module to test
+          Example: 
+            all --> Test all modules
+            diff --> Test only the modules that have changes
+            azure_api_management --> Test azure_api_management module
+        type: choice
+        required: true
+        default: diff
+        options:
+          - all
+          - diff
+          - azure_api_management
+          - azure_app_service
+          - azure_app_service_exposed
+          - azure_cosmos_account
+          - azure_event_hub
+          - azure_function_app
+          - azure_function_app_exposed
+          - azure_naming_convention
+          - azure_postgres_server
+          - azure_role_assignments
+          - azure_storage_account
+
+concurrency:
+  group: tf-test-${{ github.head_ref }}
+  cancel-in-progress: false
+
+env:
+  BASE_DIR: infra/modules/
+  ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+  ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+
+jobs:
+  tf_test:
+    name: Terraform Test
+    environment: dev-ci
+    runs-on: ubuntu-20.04
```

Comment:
it should never happen, also because the tests will all be refactored to be performed in `DEV-ENGINEERING`, and they will all be ad hoc on dx modules, I would avoid adding this possibility